elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTransport.java

```diff
@@ -118,7 +120,14 @@ private ElasticsearchVersion version() {
    * Build index mapping returning new instance of {@link ElasticsearchMapping}.
    */
   private ElasticsearchMapping fetchAndCreateMapping() {
-    final String uri = String.format(Locale.ROOT, "/%s/_mapping", indexName);
+    String encodedIndexName = indexName;
+    try {
+      encodedIndexName = URLEncoder.encode(indexName, StandardCharsets.UTF_8.toString());
+    } catch (Exception e) {
```

Comment:
Does it make sense to (re)use [URIBuilder](https://hc.apache.org/httpcomponents-client-4.3.x/httpclient/apidocs/org/apache/http/client/utils/URIBuilder.html) ?
I think it encodes path segments automatically.

Reply:
Good idea, will see how to implement.